pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -435,7 +460,12 @@ stdenv.mkDerivation {
     "-Dsmack=true"
     "-Db_pie=true"
     "-Dinstall-sysconfdir=false"
-    "-Defi-ld=${stdenv.cc.bintools.targetPrefix}ld"
+    "-Defi-ld=gold"
```

Comment:
Did this build before this PR?

Reply:
Yes. In systemd v249.7 the variable was used to invoke the linker directly (see [meson.build](https://github.com/systemd/systemd-stable/blob/v249.7/src/boot/efi/meson.build#L276)), so that's why `ld` was valid and defaulted to `bfd`.

In systemd v250.3 it's passed to `-fuse-ld` instead.

Reply:
https://github.com/NixOS/nixpkgs/pull/150491#discussion_r767351535